Compile warnings in clang 18

When the google test version bumps to the lates, we can see below warnings when we compile tests with clang 18
```
In file included from /home/runner/work/proxy/proxy/build-clang-18/_deps/googletest-src/googletest/src/gtest-all.cc:38:
In file included from /home/runner/work/proxy/proxy/build-clang-18/_deps/googletest-src/googletest/include/gtest/gtest.h:55:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/memory:66:
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_tempbuf.h:263:8: warning: 'get_temporary_buffer<testing::TestInfo *>' is deprecated [-Wdeprecated-declarations]
  263 |                 std::get_temporary_buffer<value_type>(_M_original_len));
      |                      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_algo.h:4996:15: note: in instantiation of member function 'std::_Temporary_buffer<__gnu_cxx::__normal_iterator<testing::TestInfo **, std::vector<testing::TestInfo *>>, testing::TestInfo *>::_Temporary_buffer' requested here
 4996 |       _TmpBuf __buf(__first, (__last - __first + 1) / 2);
      |               ^
/usr/bin/../lib/gcc/x[86](https://github.com/microsoft/proxy/actions/runs/10312728530/job/28548416779#step:8:87)_64-linux-gnu/12/../../../../include/c++/12/bits/stl_algo.h:5070:23: note: in instantiation of function template specialization 'std::__stable_sort<__gnu_cxx::__normal_iterator<testing::TestInfo **, std::vector<testing::TestInfo *>>, __gnu_cxx::__ops::_Iter_comp_iter<(lambda at /home/runner/work/proxy/proxy/build-clang-18/_deps/googletest-src/googletest/src/gtest.cc:3009:20)>>' requested here
 5070 |       _GLIBCXX_STD_A::__stable_sort(__first, __last,
      |                       ^
/home/runner/work/proxy/proxy/build-clang-18/_deps/googletest-src/googletest/src/gtest.cc:3008:8: note: in instantiation of function template specialization 'std::stable_sort<__gnu_cxx::__normal_iterator<testing::TestInfo **, std::vector<testing::TestInfo *>>, (lambda at /home/runner/work/proxy/proxy/build-clang-18/_deps/googletest-src/googletest/src/gtest.cc:3009:20)>' requested here
 3008 |   std::stable_sort(test_info_list_.begin(), test_info_list_.end(),
      |        ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/stl_tempbuf.h:99:5: note: 'get_temporary_buffer<testing::TestInfo *>' has been explicitly marked deprecated here
   99 |     _GLIBCXX17_DEPRECATED
      |     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/x86_64-linux-gnu/c++/12/bits/c++config.h:119:34: note: expanded from macro '_GLIBCXX17_DEPRECATED'
  119 | # define _GLIBCXX17_DEPRECATED [[__deprecated__]]
      |                                  ^
```

--- a/.github/workflows/bvt-clang.yml
+++ b/.github/workflows/bvt-clang.yml
@@ -7,7 +7,7 @@ on:
 
 jobs:
   bvt-clang:
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     steps:
     - uses: actions/checkout@v3
       with:
@@ -15,12 +15,6 @@ jobs:
 
     - name: install clang
       run: |
-        wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
-        sudo apt-add-repository "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-15 main" -y
-        sudo apt-add-repository "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-16 main" -y
-        sudo apt-add-repository "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-17 main" -y
-        sudo apt-add-repository "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-18 main" -y
-        sudo apt update
         sudo apt install -y clang-15 clang-16 clang-17 clang-18
 
     - name: check compiler versions

--- a/.github/workflows/bvt-gcc.yml
+++ b/.github/workflows/bvt-gcc.yml
@@ -7,7 +7,7 @@ on:
 
 jobs:
   bvt-gcc:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     steps:
     - uses: actions/checkout@v3
       with:
@@ -15,15 +15,14 @@ jobs:
 
     - name: install gcc
       run: |
-        sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
-        sudo apt update
-        sudo apt install -y gcc-11 g++-11 gcc-12 g++-12 gcc-13 g++-13
+        sudo apt install -y gcc-11 g++-11 gcc-12 g++-12 gcc-13 g++-13 gcc-14 g++-14
 
     - name: check compiler versions
       run: |
         g++-11 --version
         g++-12 --version
         g++-13 --version
+        g++-14 --version
 
     - name: build and run test with gcc 11
       run: |
@@ -42,3 +41,9 @@ jobs:
         cmake -B build-gcc-13 -DCMAKE_C_COMPILER=gcc-13 -DCMAKE_CXX_COMPILER=g++-13
         cmake --build build-gcc-13 -j
         ctest --test-dir build-gcc-13 -j
+
+    - name: build and run test with gcc 14
+      run: |
+        cmake -B build-gcc-14 -DCMAKE_C_COMPILER=gcc-14 -DCMAKE_CXX_COMPILER=g++-14
+        cmake --build build-gcc-14 -j
+        ctest --test-dir build-gcc-14 -j

